src/embed/init.impl.ts

```diff
@@ -120,10 +120,25 @@ export default function _dlo_initializeFromWindow() {
       Logger.getInstance().warn(LogMessageType.ObserverRulesNone);
     }
 
+    // get the log level from win._dlo_logLevel
+    // we allow string or number, and use parseInt if it is a string
+    // otherwise it is set to undefined, which will use the default log level
+    let level;
+    switch (typeof win._dlo_logLevel) {
+      case 'number':
+        level = win._dlo_logLevel;
+        break;
+      case 'string':
```

Comment:
JW, was there something we documented that indicated a string was valid?

Reply:
Yes, I did some testing and a string value that was a number actually worked.  It's in the long slack thread.

Reply:
From the thread...

It looks like whatever you set it to whether 2 or "foo" or "3" gets preserved.  It all then comes down to

```
if (level <= this.level || level === LogLevel.RECORD) {
      this.appender.log({
        level,
        message,
        context,
      });
    }
```
which does do type promotion if it applies.  If it is a string like foo it never returns true so nothing gets logged.  However, if it is a string with a number in it like "2" that does seem to get promoted.  While 0 doesn't work because of the bug, "0" does work.  :upside_down_face:.  However, I can't get around my problem as I'm using the integration, which I"m guessing most of our customers are using, and that uses a numerical value.

Reply:
I think we can make this simpler _if_ we don't want to make a decision about changing broader log level config behavior and instead want to narrowly focus on fixing this bug.

We could consider passing `win._dlo_logLevel` directly to the `DataLayerObserver` constructor:
```
    win._dlo_observer = new DataLayerObserver({
      ...
      logLevel: win._dlo_logLevel,
      ...
    });
```

Then, in the `DataLayerObserver` constructor, we could explicitly check against `undefined` instead of checking that `logLevel` is truthy (as a bonus, this is compatible with the type being `LogLevel | undefined`):
```
    if (logLevel !== undefined) {
      Logger.getInstance().level = logLevel;
    }
```

Then, we could capture the current behavior with tests. Something like the following in `observer.spec.ts`:
```
  [
    { configLevel: 0, expectedLevel: 0 },
    { configLevel: 1, expectedLevel: 1 },
    { configLevel: 2, expectedLevel: 2 },
    { configLevel: 3, expectedLevel: 3 },
    { configLevel: undefined, expectedLevel: () => Logger.getInstance().level },
  ].forEach((tc) => {
    it(`sets the global log level to ${tc.expectedLevel} given config value ${tc.configLevel}`, () => {
      const observer = ExpectObserver.getInstance().create({
        logLevel: tc.configLevel as number,
        rules: [],
      });

      const expectedLevel = typeof tc.expectedLevel === 'function' ? tc.expectedLevel() : tc.expectedLevel;
      expect(Logger.getInstance().level).to.equal(expectedLevel);

      ExpectObserver.getInstance().cleanup(observer);
    });
  });
```

This would pave the way for us to define expected string or out of bounds behaviors (e.g. `-1` or `4`) in the future without addressing those behavior changes now.

WDYT? /cc @van-fs

Reply:
That looks fine to me.  It looks like you have the code already written, should we have you just push that, or do you want me to update the code from it?

Reply:
@irv-fs Could you take Nate's suggestion and update the PR?  It sounds like the recommendations are:
1. Keep the init.impl.js shallow and directly pass the expando.
2. Handle the assignment in the DataLayerObserver constructor as-is.